### ISSUE TITLE
Support for multiple user data dirs

### ISSUE THREAD

**Issue by Vidyut** [makehumancommunity/mpfb2#130]
Currently we set the data dir in preferences.

I'd like an _additional_ way to set a supplementary dir - maybe in library settings or something (or preferences should be fine)

The usability benefit of this will be that I can manage the somewaht unwieldy (but oh so desired) assets by project and leave only the system assets in the main data dir.

That way, I can put only the assets I'm likely to need in a nimble little data directory and have everything I want without having to scroll past things I don't want (sometimes never want, never will want). Wrinkle targets for older characters, trousers for the tomboy, girly dresses for the more feminine character, long black hairstyles for the long black-haired character, curly bobs for another and so on.

Will create an ability to create a library of "look" related items by character for ongoing projects. Clothes may change, hair style may change, but the overall look of the character can be very consistent, because it is part of the design and has its own library of assets the character uses (that won't be accidentally put on others)

**Comment by joepal1976** [comment]
I don't think this should be totally impossible. In principle the system has support for multiple asset roots, but it is possible there are place where the code assumes there is only one user data directory.

**Comment by joepal1976** [comment]
In order to have a per-blend setting for this, #97 would need to be solved first. Until then the only available option would be another field in preferences and requiring a blender restart in order to discover what assets are in the extra location.

**Comment by joepal1976** [comment]
I've now pushed an update where you can set an additional asset root in the preferences:

![image](https://github.com/makehumancommunity/mpfb2/assets/10105480/f56efc79-dc2d-4fa3-a442-28223da1c7e2)

If pointing this to a directory containing (for example) a "clothes" and a "hair" subdir, and having restarted blender, contents in these dirs will be discovered together with the other assets.

So you'd have to manually move assets from the normal data directory to the secondary root, but at least you'll be able to easily toggle its visibility: just set it to blank in the preferences when you have no need of the assets in it.

@Vidyut does this solve your use case decently? I'd still need to solve #97 to be able to toggle the visibility on and off from inside blender without restarting.

**Comment by joepal1976** [comment]
While I was at it, I added a UI panel for overriding the secondary root on a per-blend basis:

![image](https://github.com/makehumancommunity/mpfb2/assets/10105480/76ccc708-cd2a-4d69-adac-5e54118dc37b)

If you change this, the asset lists will be reloaded **without** restarting blender, and the value will be persisted only for the blend file you're working on. If left empty, the value will be read from the preferences. Further I've made a change to rescan the asset lists if the preferences are changed for either the MH data dir or the secondary root. This too without restarting blender.

I think this should be tested thoroughly as it changes some of the logic on when assets are rescanned. But my preliminary impression is that it works with no visible downsides.
